ai.finto.fi/static/js/annif.js

```diff
@@ -84,45 +92,96 @@ const mainApp = createApp({
       show_alert_file_format: false,
       show_alert_request_failed: false,
       show_alert_request_failed_url: false,
+      show_alert_language_detection_failed: false,
       show_dragging_effect: false,
       supported_formats: ['txt', 'pdf', 'doc', 'docx', 'odt', 'rtf', 'pptx', 'epub', 'html']
     }
   },
+  provide() {
+    return {
+      detectLanguage: this.detectLanguage
+    };
+  },
+  watch: {
+    text_language(newValue, oldValue) {
+      if (newValue != null) {
+        this.show_alert_language_detection_failed = false;
+      }
+    },
+  },
   methods: {
     clear() {
       // clear previous text, suggestions, alerts etc
       this.text = ''
       this.show_results = false
       this.selected_file = ''
       this.selected_url = ''
+      this.is_language_detected = false;
       this.show_alert_file_size = false
       this.show_alert_file_format = false
       this.show_alert_request_failed = false
       this.show_alert_request_failed_url = false
+      this.show_alert_language_detection_failed = false
+    },
+    detectLanguage() {
+      if (this.text.length > 9) {
+        this.detecting_language = true;
+        this.text_language = 'none';
+        fetch(annif_base_url + 'detect-language', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            text: this.text,
+            languages: ["fi", "sv", "en"]  // TODO Here should be only langs that selected project supports
+          })
+        })
+        .then(response => response.json())
+        .then(data => {
+          this.text_language = data.results[0].language;
+          // this.text_language_detection_score = data.results[0].score;  TODO Add to tooltip
```

Comment:
Is the TODO a problem? Should it be fixed now instead of leaving it like this?

Reply:
Implemented this now: the language detection scores/probabilities are shown with hover-on on language selections buttons. 
![Screenshot from 2025-03-08 12-07-23](https://github.com/user-attachments/assets/16758ec2-65b0-42ec-b746-49c4ec73232b)